argo-cd-apps/base/keycloak/keycloak.yaml

```diff
@@ -0,0 +1,49 @@
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: keycloak
+spec:
+  generators:
+    - merge:
+        mergeKeys:
+          - nameNormalized
+        generators:
+          - clusters:
+              values:
+                sourceRoot: components/keycloak
+                environment: staging
+                clusterDir: base
+              selector:
+                matchLabels:
+                  appstudio.redhat.com/internal-member-cluster: "true"
+          - list:
+              elements: []
+  template:
+    metadata:
+      name: keycloak-{{nameNormalized}}
+    spec:
+      project: default
+      source:
+        path: '{{values.sourceRoot}}/{{values.environment}}/{{values.clusterDir}}'
+        repoURL: https://github.com/redhat-appstudio/infra-deployments.git
+        targetRevision: main
+      destination:
+        namespace: rhtap-auth
+        server: '{{server}}'
+      ignoreDifferences:
+        - group: keycloak.org
+          kind: KeycloakRealm
+          jsonPointers:
+            - /spec/realm/identityProviders/0/config/clientSecret
```

Comment:
this is going to be required when I add the manifest for deploying keycloak itself.